pkg/controller/job/job_controller.go

```diff
@@ -1038,8 +1040,16 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 		} else if pod.Status.Phase == v1.PodFailed || podTerminating {
 			ix := getCompletionIndex(pod.Annotations)
 			if !uncounted.failed.Has(string(pod.UID)) && (!isIndexed || (ix != unknownCompletionIndex && ix < int(*job.Spec.Completions))) {
-				needsFlush = true
-				uncountedStatus.Failed = append(uncountedStatus.Failed, pod.UID)
+				if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+					_, shouldIgnore := matchPodFailurePolicy(job, pod)
```

Comment:
If possible, avoid double negation. The variable could be called: `shouldCountFailure`

Then, it would be `if shouldCountFailure`

Otherwise, you can do `if !shouldIgnore`